checks.d/disk.py

```diff
@@ -187,7 +187,19 @@ def _collect_part_metrics(self, part, usage):
 
     def _collect_inodes_metrics(self, mountpoint):
         metrics = {}
-        inodes = os.statvfs(mountpoint)
+        # we need to timeout this, too.
+        try:
+            inodes = timeout(5)(os.statvfs)(mountpoint)
+        except TimeoutException:
+            self.log.warn(
+                u"Timeout while retrieving the disk usage of `%s` mountpoint. Skipping...",
+                mountpoint
+            )
+            return metrics
+        except Exception as e:
+            self.log.warn("Unable to get disk metrics for %s: %s", mountpoint, e)
+            return metrics
+
```

Comment:
I believe returning an empty dictionary would cause a `KeyError` exception [here](https://github.com/DataDog/dd-agent/blob/greg/statvfs-fix/checks.d/disk.py#L130).

Alternatively we could catch the exception within the `collect_metrics_psutil` method.